### PR TITLE
[13.0][IMP] Add filter on sale_cutoff_time_delivery

### PR DESCRIPTION
**Pull request by TDu**
Add cutoff filter on stock.picking views to allow filtering on picking
before yesterday's cutoff, inside today's cutoff and in future cuttofs.

### CODE REVIEW

File: sale_cutoff_time_delivery/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    cutoff_time_hms = fields.Char(
+        compute="_compute_cutoff_time_HMS", store=True, default="00:00"
```
Comment (sebalix) [inline sale_cutoff_time_delivery/models/stock_picking.py]: Could avoid this uppercase suffix?
```suggestion
        compute="_compute_cutoff_time", store=True, default="00:00"
```

File: sale_cutoff_time_delivery/models/stock_picking.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+         1 scheduled_date is in the future of today's cutoff time
+
+        """
+        for record in self:
+            warehouse = record.location_id.get_warehouse()
+            hour, minute = warehouse._get_hour_min_from_value(warehouse.cutoff_time)
+            today_cutoff = fields.Datetime.now().replace(
+                hour=hour, minute=minute, second=0
+            )
+            yesterday_cutoff = today_cutoff + relativedelta(days=-1)
```
Comment (sebalix) [inline sale_cutoff_time_delivery/models/stock_picking.py]:
```suggestion
            yesterday_cutoff = fields.Datetime.subtract(today_cutoff, days=1)
```
And you can remove your import of `relativedelta`.

File: sale_cutoff_time_delivery/models/stock_picking.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            elif record.scheduled_date > today_cutoff:
+                record.cutoff_time_diff = 1
+            else:
+                record.cutoff_time_diff = 0
+
+    def _search_cutoff_time_diff(self, operator, value):
+        if operator not in ("=", "!="):
+            raise UserError(_("Unsupported search operator %s") % (operator,))
+        today = fields.Datetime.now()
+        yesterday = today - relativedelta(days=1)
```
Comment (sebalix) [inline sale_cutoff_time_delivery/models/stock_picking.py]:
```suggestion
        yesterday = fields.Datetime.subtract(today, days=1)
```

File: sale_cutoff_time_delivery/models/stock_picking.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        params = {
+            "yesterday": yesterday.strftime("%Y-%m-%d"),
+            "today": today.strftime("%Y-%m-%d"),
```
Comment (sebalix) [inline sale_cutoff_time_delivery/models/stock_picking.py]: As the format expected by Pg is already defined in Odoo internals (https://github.com/odoo/odoo/blob/13.0/odoo/tools/misc.py#L584), you could do:
```suggestion
            "yesterday": fields.Date.to_string(yesterday),
            "today": fields.Date.to_string(today),
```

File: sale_cutoff_time_delivery/models/stock_picking.py (hunk 1 above)
Comment (TDu) [reply sale_cutoff_time_delivery/models/stock_picking.py]: Renamed it but kept the `hms` in lower case.

File: sale_cutoff_time_delivery/models/stock_picking.py (hunk 2 above)
Comment (TDu) [reply sale_cutoff_time_delivery/models/stock_picking.py]: Oh, Odoo internals to the rescue, changed.

File: sale_cutoff_time_delivery/models/stock_picking.py (hunk 3 above)
Comment (TDu) [reply sale_cutoff_time_delivery/models/stock_picking.py]: Changed

File: sale_cutoff_time_delivery/models/stock_picking.py (hunk 4 above)
Comment (TDu) [reply sale_cutoff_time_delivery/models/stock_picking.py]: Ok, I changed it.

**Review by sebalix — APPROVED** [review]
Technical review only

**Comment by simahawk** [comment]
@TDu please close this and open it on OCA as the base module is merged now.

**Comment by TDu** [comment]
Move to the OCA

**Pull request closed without merging**